src/mlpack/methods/cf/normalization/user_mean_normalization.hpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file user_mean_normalization.hpp
+ * @author Wenhao Huang
+ *
+ * This class performs user mean normalization on raw ratings. In another
+ * word, this class is used to remove global effect of user mean.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_CF_NORMALIZATION_USER_MEAN_NORMALIZATION_HPP
+#define MLPACK_METHODS_CF_NORMALIZATION_USER_MEAN_NORMALIZATION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace cf {
+
+/**
+ * This normalization class performs user mean normalization on raw ratings.
+ *
+ * An example of how to use UserMeanNormalization in CF is shown below:
+ *
+ * @code
+ * extern arma::mat data; // data is a (user, item, rating) table.
+ * // Users for whom recommendations are generated.
+ * extern arma::Col<size_t> users;
+ * arma::Mat<size_t> recommendations; // Resulting recommendations.
+ *
+ * // Use UserMeanNormalization as normalization method.
+ * CFType<UserMeanNormalization> cf(data);
+ *
+ * // Generate 10 recommendations for all users.
+ * cf.GetRecommendations(10, recommendations);
+ * @endcode
+ */
+class UserMeanNormalization
+{
+ public:
+  // Empty constructor.
+  UserMeanNormalization() { }
+
+  /**
+   * Normalize the data by subtracting user mean from each of existing ratings.
+   *
+   * @param data Input dataset in the form of coordinate list.
+   */
+  void Normalize(arma::mat& data)
+  {
+    const size_t userNum = arma::max(data.row(0)) + 1;
+    userMean = arma::vec(userNum, arma::fill::zeros);
```

Comment:
You should replace the datatype by `arma::rowvec` since actually it is a row. That should fix the Travis issue.